doc/plugins.dox

```diff
@@ -170,14 +171,22 @@ Geany's website at http://www.geany.org/Support/BuildingOnWin32.
 
 @section helloworld "Hello World"
 
-When writing a plugin, you will find a couple of functions or macros which are mandatory
-and some which are free to use for implementing some useful feature once your plugin
-becomes more powerful like including a configuration or help dialog.
+@note This section describes the new entry points for plugins introduced with Geany 1.26. A short
+summary of the legacy entry points is given at page @ but it should not be used any more.
+
+When writing a plugin, you will find a couple of functions which are mandatory and some which are
+free to use for implementing some useful feature once your plugin becomes more powerful like
```

Comment:
Also, I'm not fond of the formulation, something more formal would IMO be clearer, like I don't know,

"Some functions are required in every plugin, and some are optional but useful to implement some more advanced features, like a help button or a configuration dialog."

Or something.  Not sure if this sentence is in the right place though.

This said, as @elextr said probably the whole howto would benefit from a lifting, so it might be OK-ish to have something imperfect so long as it's clear enough.